aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -709,7 +709,7 @@ void launch(
   }
 
 #if CUB_SUPPORTS_SCAN_BY_KEY()
-  computeBlockwiseKthCounts<Bitwise><<<std::min((numInputSlices + 255) / 256, (uint32_t)65535), 256, 0, stream>>>(
+  computeBlockwiseKthCounts<Bitwise><<<std::min(((int64_t)numInputSlices + 255) / 256, (int64_t)1073741824), 256, 0, stream>>>(
```

Comment:
Well, realistically memory that's enough to hold 256 G slices won't appear in our lifetimes, but ok.